modules/tinymce/src/core/main/ts/keyboard/PageUpDownKeys.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Cell } from '@ephox/katamari';
+import Editor from '../api/Editor';
+import { NodeChangeEvent } from '../api/EventTypes';
+import { EditorEvent } from '../api/PublicApi';
+import VK from '../api/util/VK';
+import * as InlineBoundariesNavigation from './InlineBoundariesNavigation';
+import * as MatchKeys from './MatchKeys';
+
+const executeKeyupAction = (editor: Editor, evt: KeyboardEvent) => {
+  MatchKeys.execute([
+    { keyCode: VK.PAGE_UP, action: MatchKeys.action(InlineBoundariesNavigation.moveOutInlineBoundaries, editor, false) },
+    { keyCode: VK.PAGE_DOWN, action: MatchKeys.action(InlineBoundariesNavigation.moveOutInlineBoundaries, editor, true) }
+  ], evt);
+};
+
+const stopImmediatePropagation = (e: EditorEvent<NodeChangeEvent>) => e.stopImmediatePropagation();
+
+const isPageUpDown = (evt: EditorEvent<KeyboardEvent>) => {
+  return evt.keyCode === VK.PAGE_UP || evt.keyCode === VK.PAGE_DOWN;
+};
+
+const setNodeChangeBlocker = (blocked: Cell<boolean>, editor: Editor, block: boolean) => {
```

Comment:
This logic doesn't appear to be working when testing locally, as when pressing page down I'm still seeing a blue flash whereby the inline boundary applies quickly and then is removed.

I haven't had a chance to debug it, but that's not ideal and we should see if we can fix it.

Reply:
Okay, I looked into this and it's because you're calling `editor.nodeChanged()` before moving the selection. If you update the below to be something like this, then it should resolve that:
```
  editor.on('keyup', (evt) => {
    if (evt.isDefaultPrevented() === false) {
      executeKeyupAction(editor, evt);
    }
    if (isPageUpDown(evt)) {
      setNodeChangeBlocker(blocked, editor, false);
      editor.nodeChanged();
    }
  });
```

Reply:
updated